tests/test_subset.py

```diff
@@ -42,10 +45,16 @@ def test_subset_granule(maap: MAAP, h5_path: str, aoi_gdf: gpd.GeoDataFrame):
             aoi_gdf,
             "lat_lowestmode",
             "lon_lowestmode",
+            "all",
             ["agbd"],
             "l2_quality_flag == 1",
-            output_dir,
+            Path(output_dir),
         )
     )
 
     assert io_result == IOSuccess(Some(expected_path))
+
+
+def test_bad_check_beams_option(value="0100"):
+    with pytest.raises(BadParameter):
+        check_beams_option(value)
```

Comment:
Awesome job on using `pytest.raises`!

Let's keep going with more tests, both for invalid values as well as valid values.

First, let's adjust this specific test. Normally, since `value` is _not_ supplied by a pytest fixture, it should _not_ be a function parameter. However, you can keep it there _if_ you parameterize it, like we did for our `test_subset_hdf5` function (but simpler here, as we need only 1 parameter).

Next, add a `test_valid_check_beams_option` function parameterized with many valid values, including varying upper/lowercase values. For parameters on this "valid" test function, you'll need 2 parameters: (1) input value (varying upper/lower, etc.), and (2) expected value (normalized value, which should be all uppercase, and for comma-separated values, all with BEAM prefixes).